### PR TITLE
[TOOLS-235] Export transitions - Data

### PR DESCRIPTION
**Pull request by jhonypg**

### CODE REVIEW

File: src/main/java/objective/taskboard/followup/impl/FollowUpTransitionsDataProvider.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+            for (DateTime date : row.transitionsDates) {
+                if (date == null)
+                    continue;
+                if (firstDate == null || date.isBefore(firstDate))
+                    firstDate = date;
+                if (lastDate == null || date.isAfter(lastDate))
+                    lastDate = date;
+            }
+        }
+        return new Interval(firstDate.dayOfMonth().roundFloorCopy(), lastDate.plusDays(1));
```
Comment (objective-solutions-user) [inline src/main/java/objective/taskboard/followup/impl/FollowUpTransitionsDataProvider.java]: MAJOR A "NullPointerException" could be thrown; "firstDate" is nullable here. [rule](http://10.44.1.250:9000/coding_rules#rule_key=squid%3AS2259)

**Comment by objective-solutions-user** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: src/test/java/objective/taskboard/followup/impl/FollowUpTransitionsDataProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+                    .transition("To Review", "2017-01-02")
+                    .transition("Reviewing", "2017-01-02")
+                    .transition("Done", "2017-01-02").issueStatus(statusDone)
+                , subtask().id(102).key("PROJ-102").startDate(parseDate("2017-01-03"))
+                    .transition("To Do", "2017-01-04")
+                    .transition("Doing", "2017-01-05")
+                    .transition("To Review", "2017-01-06")
+                    .transition("Reviewing", "2017-01-07")
+                    .transition("Done", "2017-01-08").issueStatus(statusDone)
+                , subtask().id(103).key("2017-01-03").startDate(parseDate("2017-01-04"))
```
Comment (jytakano) [inline src/test/java/objective/taskboard/followup/impl/FollowUpTransitionsDataProviderTest.java]: key("2017-01-03")

File: src/main/java/objective/taskboard/followup/impl/FollowUpTransitionsDataProvider.java (hunk 1 above)
Comment (jytakano) [reply src/main/java/objective/taskboard/followup/impl/FollowUpTransitionsDataProvider.java]: Seria bom um código defensivo

**Pull request merged**